src/Auth/Proofs/Phrase.php

```diff
@@ -0,0 +1,57 @@
+
+    private array $nouns = ['apple', 'banana', 'cat', 'dog', 'elephant', 'fish', 'guitar', 'hat', 'ice cream', 'jacket', 'kangaroo', 'lemon', 'moon', 'notebook', 'orange', 'piano', 'quilt', 'rabbit', 'sun', 'tree', 'umbrella', 'violin', 'watermelon', 'xylophone', 'yogurt', 'zebra', 'airplane', 'ball', 'cloud', 'diamond', 'eagle', 'fire', 'giraffe', 'hammer', 'island', 'jellyfish', 'kiwi', 'lamp', 'mango', 'needle', 'ocean', 'pear', 'quasar', 'rose', 'star', 'turtle', 'unicorn', 'volcano', 'whale', 'xylograph', 'yarn', 'zephyr', 'ant', 'book', 'candle', 'door', 'envelope', 'feather', 'globe', 'harp', 'insect', 'jar', 'kite', 'lighthouse', 'magnet', 'necklace', 'owl', 'puzzle', 'queen', 'rainbow', 'sailboat', 'telescope', 'umbrella', 'vase', 'wallet', 'xylograph', 'yacht', 'zeppelin', 'accordion', 'brush', 'chocolate', 'dolphin', 'easel', 'fountain', 'globe', 'hairbrush', 'iceberg', 'jigsaw', 'kettle', 'leopard', 'marble', 'nutmeg', 'obstacle', 'penguin', 'quiver', 'raccoon', 'sphinx', 'trampoline', 'utensil', 'velvet', 'wagon', 'xerox', 'yodel', 'zipper'];
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct('argon2', ['argon2' => new Argon2()]);
+    }
+
+    /**
+     * Generate a proof based on input
+     *
+     * @param  string  $input Input to generate proof from (unused in this implementation)
+     * @return string
+     */
+    public function generate(string $input): string
+    {
+        $adjective = $this->adjectives[array_rand($this->adjectives)];
+        $noun = $this->nouns[array_rand($this->nouns)];
+
+        return "{$adjective} {$noun}";
+    }
+
+    /**
+     * Hash a proof
+     *
+     * @param  string  $proof
+     * @return string
+     */
+    public function hash(string $proof): string
+    {
+        return password_hash($proof, PASSWORD_DEFAULT);
+    }
+
+    /**
+     * Verify a proof
+     *
+     * @param  string  $proof
+     * @param  string  $hash
+     * @return bool
+     */
+    public function verify(string $proof, string $hash): bool
+    {
+        return password_verify($proof, $hash);
```

Comment:
This is not using algorithm class; if ago was md5, this would still be using argon